libs/vkd3d/device.c

```diff
@@ -1350,12 +1353,20 @@ static HRESULT vkd3d_init_device_caps(struct d3d12_device *device,
         device->feature_options.TiledResourcesTier = D3D12_TILED_RESOURCES_TIER_NOT_SUPPORTED;
     }
 
-    if (device->vk_info.device_limits.maxPerStageDescriptorSamplers <= 16)
-        device->feature_options.ResourceBindingTier = D3D12_RESOURCE_BINDING_TIER_1;
-    else if (device->vk_info.device_limits.maxPerStageDescriptorUniformBuffers <= 14)
+    /* Only advertize Tier 2 on hardware that supports bindless */
+    device->feature_options.ResourceBindingTier = D3D12_RESOURCE_BINDING_TIER_1;
+
+    /* Ignore sampler count since the minimum required
+     * number of samplers to support per set is 500k */
+    if (descriptor_indexing->runtimeDescriptorArray
+            && descriptor_indexing->shaderSampledImageArrayNonUniformIndexing
+            && descriptor_indexing->shaderUniformTexelBufferArrayNonUniformIndexing
+            && descriptor_indexing->descriptorBindingSampledImageUpdateAfterBind
+            && descriptor_indexing->descriptorBindingUpdateUnusedWhilePending
+            && descriptor_indexing->descriptorBindingPartiallyBound
+            && descriptor_indexing->descriptorBindingVariableDescriptorCount
+            && descriptor_indexing_properties->maxPerStageDescriptorUpdateAfterBindSampledImages >= 1000000)
         device->feature_options.ResourceBindingTier = D3D12_RESOURCE_BINDING_TIER_2;
```

Comment:
We'll probably need to report TIER_3 here anyways because WoW is broken and refuses to use D3D12 unless that is reported. Is there an "app-opt" thing in place we can use for this?

Reply:
> Is there an "app-opt" thing in place we can use for this?

No. I guess the best thing to do for now is to keep the old code for this and instead check for bindless features in `vkd3d_bindless_state_get_bindless_flags`.

Reply:
That works, please add comments explaining the situation in that case.

Reply:
Done.